app/models/job_invocation_composer.rb

```diff
@@ -0,0 +1,217 @@
+class JobInvocationComposer
+  attr_accessor :params, :job_invocation, :host_ids, :search_query
+  attr_reader :job_template_ids
+
+  # TODO aka initialize_from_params
+  def initialize(job_invocation, params)
+    @job_invocation = job_invocation
+    @params = params
+
+    @host_ids = validate_host_ids(params[:host_ids])
+    @search_query = targeting_base[:search_query]
+
+    job_invocation.job_name = validate_job_name(job_invocation_base[:job_name]) || available_job_names.first
```

Comment:
I might be misunderstanding this line but is this saying use the specified job_name if available or just grab the first one the user can read?   I guess this makes sense for the 'new' case, but seems odd if there's an 'update' that goes wrong